Confidence/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt

```diff
@@ -118,29 +118,34 @@ class EventSenderIntegrationTest {
                 return false
             }
         }
+        val debugLogger = DebugLoggerMock()
         val engine = EventSenderEngineImpl(
             eventStorage,
             clientSecret,
             flushPolicies = mutableListOf(flushPolicy),
             dispatcher = testDispatcher,
             sdkMetadata = SdkMetadata("kotlin_test", ""),
-            uploader = uploader
+            uploader = uploader,
+            debugLogger = debugLogger
         )
         eventSender = Confidence(
             eventSenderEngine = engine,
             dispatcher = testDispatcher,
             diskStorage = mock(),
             clientSecret = "",
             flagApplierClient = mock(),
-            flagResolver = mock()
+            flagResolver = mock(),
+            debugLogger = debugLogger
         )
         val eventSender = this@EventSenderIntegrationTest.eventSender
         val eventCount = 4 * batchSize + 2
         requireNotNull(eventSender)
         repeat(eventCount) {
             eventSender.track("navigate")
         }
+        Assert.assertEquals(eventCount * 4, debugLogger.eventsLogged)
```

Comment:
what is event count * 4?

Reply:
so eventsLogged isn't a unique event, we log events when they get sent/stored/etc, so all in all we have 4 different times/places where we log events, thus *4.

Reply:
imo those etc are very important to know what we are testing. what if we change the policy numbers to one more but we have a bug that leads to not send another log message? then the count of number of logs is still 4 but the tests are passing which has to fail instead. so it's important to know what we are asserting. 
currently we log events for : emitting, writing to disk, flush policies hit (i'm also sceptical about logging this since it's not really related to any event lifecycle and it's pure implementation details, also the message is misleading since the flush policy is not really triggered for the event as it's stated in the message). is there more?

Reply:
my suggestion is to at least comment where exactly these 4 logs are coming from.

Reply:
sounds good! i had this comment previously, but wasn't sure if it should be in the tests. adding it back with elaboration